pkg/apiserver/profiling/model.go

```diff
@@ -84,11 +116,15 @@ func NewTask(ctx context.Context, taskGroup *TaskGroup, target model.RequestTarg
 }
 
 func (t *Task) run() {
-	fileNameWithoutExt := fmt.Sprintf("profiling_%d_%d_%s", t.TaskGroupID, t.ID, t.Target.FileName())
-	protoFilePath, rawDataType, err := profileAndWritePprof(t.ctx, t.fetchers, &t.Target, fileNameWithoutExt, t.taskGroup.ProfileDurationSecs)
+	fileNameWithoutExt := fmt.Sprintf("profiling_%d_%d_%s_%s", t.TaskGroupID, t.ID, t.ProfilingType, t.Target.FileName())
+	protoFilePath, rawDataType, err := profileAndWritePprof(t.ctx, t.fetchers, &t.Target, fileNameWithoutExt, t.taskGroup.ProfileDurationSecs, t.ProfilingType)
 	if err != nil {
-		t.Error = err.Error()
-		t.State = TaskStateError
+		if errorx.IsOfType(err, ErrTaskSikpped) {
```

Comment:
From the "profileAndWritePprof" perspective, it should be some kind of unsupported profiling type error, which then result in the task to be skipped.

```suggestion
		if errorx.IsOfType(err, errUnsupportedProfilingType) {
```